### ISSUE TITLE
Upgrade and maintain version for Godot 4.0

### ISSUE THREAD

**Issue by maxim-sheronov** [GameAnalytics/GA-SDK-GODOT#25]
That would be great to upgrade sdk (possibly migrate to GDExtension) and maintain it for upcoming Godot 4.0

**Comment by anthonyirwin82** [comment]
I agree godot 4 is officially released as stable now. I compiled the web export template and added the gameanalytics.js and modified the HTML to reference it but it does not work with godot 4.

**Comment by sabvdf** [comment]
I would expect if they follow market trends, that they pick this up by now, but maybe they're not that aware of Godot's explosive growth since Unity lost the trust of many, and Godot 4(.2) being brilliant.
I'm going to try compiling a GDExtension myself, since GA seems to be the only proper solution compared to other analytics solutions, especially coming from Unity Analytics.

**Comment by sabvdf** [comment]
For my progress on a working 4.2 version (undocumented and WIP!) you could check https://github.com/virtualplaynl/GA-SDK-GODOT/tree/gdextension

**Comment by farfalk** [comment]
With Godot at 4.2.1 stable and an incoming 4.3, it seems like a good idea to port this to the Godot 4.x GDExtension system. Nice job with your experiment @sabvdf !

**Comment by sabvdf** [comment]
It's definitely not just an experiment anymore, really going towards a fully production usable add-on:

- Got GA fully working in editor and exports in macOS, no weird tricks required, only adding a folder :)
- For Windows it should also work already, just haven't tested yet.
- Now working on iOS, which seems to be working very well except for a logging crash bug I've run into in the GA iOS SDK. Contacting GA via the issues form, hopefully they'll be quick to reply.

**Comment by farfalk** [comment]
> It's definitely not just an experiment anymore, really going towards a fully production usable add-on:
> 
>     * Got GA fully working in editor and exports in macOS, no weird tricks required, only adding a folder :)
> 
>     * For Windows it should also work already, just haven't tested yet.
> 
>     * Now working on iOS, which seems to be working very well except for a logging crash bug I've run into in the GA iOS SDK. Contacting GA via the issues form, hopefully they'll be quick to reply.

That's very nice to know! I'll try to check it out asap. Do you think it could become a PR for this project?

**Comment by sabvdf** [comment]
> That's very nice to know! I'll try to check it out asap. Do you think it could become a PR for this project?

I have set it up as a fork, it's quite a different branch by now but I would think it make sense to replace the current contents. That's up to GA / dorin to review (and it's not a small thing to review, for all 6 platforms plus GDScript / Dotnet for all but web, so I'm guessing that would take a little bit of time after I finish).

**Comment by dorin-ga** [comment]
Fixed in v3.0.0
